http-clients/src/main/java/com/palantir/remoting2/clients/ClientConfigurations.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.remoting2.clients;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.net.HostAndPort;
+import com.palantir.remoting.api.config.service.ProxyConfiguration;
+import com.palantir.remoting.api.config.service.ServiceConfiguration;
+import com.palantir.remoting2.config.ssl.SslSocketFactories;
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.net.Proxy;
+import java.net.ProxySelector;
+import java.net.SocketAddress;
+import java.net.URI;
+import java.time.Duration;
+import java.util.List;
+import java.util.Optional;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.X509TrustManager;
+
+/** Utilities for creating {@link ClientConfiguration} instances. */
+public final class ClientConfigurations {
+
+    // Defaults for parameters that are optional in ServiceConfiguration.
+    private static final Duration DEFAULT_CONNECT_TIMEOUT = Duration.ofSeconds(10);
+    private static final Duration DEFAULT_READ_TIMEOUT = Duration.ofMinutes(10);
+    private static final Duration DEFAULT_WRITE_TIMEOUT = Duration.ofMinutes(10);
+    private static final boolean DEFAULT_ENABLE_GCM_CIPHERS = false;
+    private static final int DEFAULT_MAX_NUM_RETRIES = 0;
+
+    private ClientConfigurations() {}
+
+    /**
+     * Creates a new {@link ClientConfiguration} instance from the given {@link ServiceConfiguration}, filling in
+     * empty/absent configuration with the defaults specified as constants in this class.
+     */
+    public static ClientConfiguration of(ServiceConfiguration config) {
```

Comment:
is this the only way for me to create a `ClientConfiguration` that overrides things like the timeouts and proxy?